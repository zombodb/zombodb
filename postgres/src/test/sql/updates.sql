set datestyle to 'iso, mdy';

VACUUM so_posts;

SELECT assert(count(*), 165240, '# of posts') FROM so_posts;

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert(count(*), 165240, '# of posts') FROM so_posts;
SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');
COMMIT;

SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert(count(*), 165240, '# of posts') FROM so_posts;
SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');
ABORT;

SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');

