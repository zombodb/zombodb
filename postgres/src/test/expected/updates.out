set datestyle to 'iso, mdy';
VACUUM so_posts;
SELECT assert(count(*), 165240, '# of posts') FROM so_posts;
 assert 
--------
 t
(1 row)

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert(count(*), 165240, '# of posts') FROM so_posts;
 assert 
--------
 t
(1 row)

SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');
 assert 
--------
 t
(1 row)

COMMIT;
SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');
 assert 
--------
 t
(1 row)

BEGIN;
UPDATE so_posts SET id = id WHERE id IN (SELECT id FROM so_posts ORDER BY random() LIMIT 1000);
SELECT assert(count(*), 165240, '# of posts') FROM so_posts;
 assert 
--------
 t
(1 row)

SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');
 assert 
--------
 t
(1 row)

ABORT;
SELECT assert(zdb_estimate_count('so_posts', ''), (SELECT count(*) FROM so_posts), 'estimated count');
 assert 
--------
 t
(1 row)

